storage/database.go

```diff
@@ -292,32 +303,17 @@ func (d *db) ongoingTick() {
 }
 
 func (d *db) splayedTick() {
-	shards := d.getOwnedShards()
-	if len(shards) == 0 {
+	namespaces := d.getOwnedNamespaces()
+	if len(namespaces) == 0 {
 		return
 	}
 
-	splayApart := d.tickDeadline / time.Duration(len(shards))
-
 	start := d.nowFn()
 
-	var wg sync.WaitGroup
-	for i, shard := range shards {
-		i := i
-		shard := shard
-		if i > 0 {
-			time.Sleep(splayApart)
-		}
-		wg.Add(1)
-		go func() {
-			// TODO(r): instrument timing of this tick
-			shard.Tick()
-			wg.Done()
-		}()
+	for _, n := range namespaces {
+		// TODO(xichen): sleep during two consecutive ticks
```

Comment:
Hmm how do we fairly splay here?  Wonder if we can keep a count of the IDs in each namespace then fairly divide the splay between them at the top level and tell each namespace what their deadline is.  We could potentially do this in a later diff.